src/world/particles.js

```diff
@@ -0,0 +1,96 @@
+import { on } from 'events'
+import fs from 'fs'
+
+import Vec3 from 'vec3'
+import mdata from 'minecraft-data'
+import { aiter } from 'iterator-helper'
+import combineAsyncIterators from 'combine-async-iterators'
+
+import logger from '../logger.js'
+import { Context, World } from '../events.js'
+import { abortable } from '../iterator.js'
+
+const mcData = mdata('1.16.5')
+const log = logger(import.meta)
+export default {
+  /** @type {import('../context.js').Observer} */
+  observe({ events, dispatch, client, world, signal }) {
+    aiter(
+      abortable(
+        // @ts-ignore
+        combineAsyncIterators(
+          on(events, Context.MOB_DAMAGE, { signal })
+          // on(events, Context.MOB_DEATH, { signal }),
+        )
+      )
+    )
+      .map(([{ mob }]) => ({ mob }))
```

Comment:
Here the extraction is not needed

```suggestion
      .map(([payload]) => payload)
```